### PR TITLE
Add guide for new editor

### PR DESCRIPTION
**Pull request by argenos**
## Added

Guide for the GitHub.dev editor

## Checklist
- [x] before creating a new note, I searched the vault
- [x] new notes have the `.md` extension
- [x] (if applicable) attached images have descriptive file names
- [ ] (if applicable) for new notes in the folder "04 - Guides, Workflows, & Courses", I added a link to them in one of the "for {group X}" overviews: https://publish.obsidian.md/hub/04+-+Guides%2C+Workflows%2C+%26+Courses/%F0%9F%97%82%EF%B8%8F+04+-+Guides%2C+Workflows%2C+%26+Courses

### CODE REVIEW

**Comment by argenos** [comment]
@chrisgrieser @claremacrae @kometenstaub I didn't have much time to write this down and probably won't have much time to look at this till next month. Hope this is a good starting place, but feel free to edit and/or push to this branch if you have suggestions or improvements

**Comment by argenos** [comment]
From Discord, this line in CONTRIBUTING.md can be removed:
 
> ==Before submission, remember to revert the Community Vault to its basic settings, e.g. remove installed plugins, turn on spellchecking, and change any other unusual settings back to how they previously were.==

And we need to update CONTRIBUTING to point to this guide accordingly, and any other "for X" guides if they apply.

File: Editing notes using the github.dev editor.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# Editing notes using the github.dev editor
+
+%% Add a description below this line. It doesn't need to be long: one or two sentences should be a good start. %%
+
+This guide will cover the basics of using the github.dev editor for the [[Obsidian Hub]]. This editor is great when you are making quick edits that don't need any Obsidian-specific functionality.
+
+## Opening the github.dev editor
+
+To open the editor you'll usually have two options: 
+1. On the Hub's publish site, each note has an "Edit in GitHub" button in it's footer
```
Comment (claremacrae) [inline Editing notes using the github.dev editor.md:18]: Maybe worth linking to the Hub publish site and the Hub repo from these two bullet points... Many will not know what they mean, I feel.

File: Editing notes using the github.dev editor.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+# Editing notes using the github.dev editor
+
+%% Add a description below this line. It doesn't need to be long: one or two sentences should be a good start. %%
+
+This guide will cover the basics of using the github.dev editor for the [[Obsidian Hub]]. This editor is great when you are making quick edits that don't need any Obsidian-specific functionality.
+
+## Opening the github.dev editor
+
+To open the editor you'll usually have two options:
```
Comment (claremacrae) [inline Editing notes using the github.dev editor.md:17]: I'm guessing people need to have a GitHub account, and be logged in to GitHub in their browser, for this to work???

File: Editing notes using the github.dev editor.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+6. GitHub will let you know it will switch now to the copy of the repository in your account. And once it reloads you'll see that your changes have been saved.
+	![[github.dev switch to fork.png|300]]
+	![[github.dev fork view.png]]
+
+## Submitting changes to the Hub
+
+Using git terminology, this is called creating a **Pull Request**.
+
+1. Clicking on [this](https://github.com/obsidian-community/obsidian-hub/compare) link is equivalent to going to the Hub repository, clicking on the Pull Requests tab and choosing New. Click on compare across forks, and choose yours on head repository. For the branch, switch to the one you chose on [[#^8cdfd9|step 6]] on the previous section:
+	![[github compare forks.png]]
```
Comment (claremacrae) [inline Editing notes using the github.dev editor.md:43]: The info in the pictures is great. I think we should review them after the page is published, for readability and accessibility.

And if we do re-do the images with smaller windows and large font sizes, maybe we could use files from the Hub, for context?

Here's an example of an existing page, on the Publish site, so show the type of problem:

image

**Review by claremacrae — COMMENTED** [review]
Many thanks for doing this!

I'll make a few suggested edits and push to the branch, each as separate commits...

I think it's worth merging this soon, but before too long, I think it will be worth redoing the images so that windows are smaller and the text in them is much larger - so that the text will be legible in the publish site. Especially when it's white text on a black background, which makes shrunk text even harder to read...

File: Editing notes using the github.dev editor.md (hunk 1 above)
Comment (claremacrae) [reply Editing notes using the github.dev editor.md:18]: I've added those links.

File: Editing notes using the github.dev editor.md (hunk 2 above)
Comment (claremacrae) [reply Editing notes using the github.dev editor.md:17]: I've added a note about that.

**Comment by claremacrae** [comment]
I've pushed a few suggested additions - and would be grateful for review.

File: Editing notes using the github.dev editor.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+publish: true
+---
+
+# Editing notes using the github.dev editor
+
+%% Add a description below this line. It doesn't need to be long: one or two sentences should be a good start. %%
+
+This guide will cover the basics of using the github.dev editor for the [[Obsidian Hub]]. This editor is great when you are making quick edits that don't need any Obsidian-specific functionality.
+
+In particular, if you are browsing the Hub and you see a note that you would like to improve, this guide talks you through how to make the edit, using the "Edit in GitHub" button at the bottom of the note.
```
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:15]: the guide "talks you through" sounds somewhat weird in my ears, but I am not sure how I'd rephrase it. Maybe sth like "will show you"?

File: Editing notes using the github.dev editor.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Note that you will need a GitHub account to make edits, and if you aren't logged in to it, launching the github.dev editor will prompt you for your GitHub login details.
+
+## Making changes to the notes
+
+1. To make changes, you can select files from the file explorer (1), and make edits to the note (2) as you would on any text editor or paste any changes made in Obsidian to existing notes. To add new files, you can also drag from your system's file explorer to the file explorer in the editor (1).
+	![[github.dev editor.png]]
+2. Once you are ready to submit your changes, switch to the version control tab. You'll see the names of the files you've modified under the "Changes" section:
+    ![[github.dev editor version control.png|300]]
+3. Add the files you want to submit by pressing the <kbd>+</kbd> sign (2), and write a message describing the changes you've made(1):
```
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:31]: Like, I know that there is sometimes a reason for not adding everything to a commit when working with git in general. But for the occasional contributor not familiar with git, would there ever be a scenario where you do make changes that you do *not* end up adding to the commit?

I think it would be easier to simply phrase instruct people to simply add everything (or use such an add everything button, if there is one).

File: Editing notes using the github.dev editor.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+1. To make changes, you can select files from the file explorer (1), and make edits to the note (2) as you would on any text editor or paste any changes made in Obsidian to existing notes. To add new files, you can also drag from your system's file explorer to the file explorer in the editor (1).
+	![[github.dev editor.png]]
+2. Once you are ready to submit your changes, switch to the version control tab. You'll see the names of the files you've modified under the "Changes" section:
+    ![[github.dev editor version control.png|300]]
+3. Add the files you want to submit by pressing the <kbd>+</kbd> sign (2), and write a message describing the changes you've made(1): 
+	![[github.dev editor stage changes.png|300]]
+4. To submit your files, you have to press the <kbd>✓</kbd> button at the top. 
+	The first time you make an edit to the hub, you'll get a message explaining that GitHub will create a copy of the repository on your account. This is called a **Fork**.
+	![[github.dev editor commit changes.png]]
+5. Next, it will ask you to give a name to the *branch* where you want to save your changes. You can choose a new name for it, or just press enter. ^8cdfd9
```
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:36]: for the less git experienced, I'd probably add/simplify this. Particularly for small contributions, the name the new branch gets is pretty much irrelevant, and the instruction "you can choose a new name" adds unnecessary complexity.

File: Editing notes using the github.dev editor.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+	![[github.dev switch to fork.png|300]]
+	![[github.dev fork view.png]]
+
+At this stage, your edits are in your fork of the Hub.
+
+## Submitting changes to the Hub
+
+Now you need to submit the changes in Fork to the original Hub repository.
+
+Using git terminology, this is called creating a **Pull Request**.
```
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:48]: aren't Pull requests strictly speaking git*hub* terminology? :P

File: Editing notes using the github.dev editor.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+2. Once you have selected your fork and branch, you can choose <kbd>Create pull request</kbd>
+	![[github create pull request.png]] 
+3. Fill out the template and when you are done, press on <kbd>Create pull request</kbd>.  
+	![[github fill out PR template.png]]
+
+## What happens next?
+
+Your Pull Request will be added to the [list of Pull Requests](https://github.com/obsidian-community/obsidian-hub/pulls) in the original Hub repository.
+
+Your edits will later be reviewed by a volunteer of the Hub project, and then merged in to the latest version of the Hub.
```
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:61]: you might also add that the reviewer might request changes before approval

File: Editing notes using the github.dev editor.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+Now you need to submit the changes in Fork to the original Hub repository.
+
+Using git terminology, this is called creating a **Pull Request**.
+
+1. Clicking on [this](https://github.com/obsidian-community/obsidian-hub/compare) link is equivalent to going to the Hub repository, clicking on the Pull Requests tab and choosing New. Click on compare across forks, and choose yours on head repository. For the branch, switch to the one you chose on [[#^8cdfd9|step 6]] on the previous section:
+	![[github compare forks.png]]
+2. Once you have selected your fork and branch, you can choose <kbd>Create pull request</kbd>
+	![[github create pull request.png]] 
+3. Fill out the template and when you are done, press on <kbd>Create pull request</kbd>.
```
Comment (chrisgrieser) [inline Editing notes using the github.dev editor.md:54]: for the less git experienced, it will be confusing why you have to describe your changes again, after you already described your changes above in the commit message 🤔

**Review by chrisgrieser — CHANGES_REQUESTED** [review]
I have added some comments, and while the guide in itself is very good, I am somewhat unsure about the purpose of recommending the github.dev editor to contributors.

Like this guide tells me *how* to use the github.dev editor, but it doesn't tell me *why* I should do it.

Like, using the normal GitHub Editor looks more intuitive to less tech-savvy users to me, and the other guide you wrote for that also contains far less references to git concepts (fork, branch, etc.). After reading your guide I am still asking myself, what the benefit of using the github.dev editor actually is. There are some things like having code editor commands when editing, but a less tech-savvy contributor might not be missing those anyway, but is confronted with a more complex editor with a lot more buttons and git lingo. 

And for the more tech-savvy contributors, using the githubdev editor means potentially learning a new tool when they are using a different code editor than VS Code (like me), as opposed to using the normal GitHub editor tech-savvy people are already familiar with. And having things like code editor commands isn't really a plus, since tech-savvy people that do more than mini-contributions would/could also simply use Obsidian (another tool they are already familiar with) and push the vault.

So I haven't followed the discussion you guys had a while back on the advantages of using the github.dev over normal github or over Obsidian, and I assume there are a bunch (otherwise you wouldn't recommend using it), but those benefits of using github.dev should be clearly emphasized at the beginning, so the reader knows why it's worth learning how to use github.dev.

**Comment by argenos** [comment]
I don’t think the text itself is necessary for all the images, it’s more about where the buttons are. I did zoom into the ones I thought mattered. 

I was not using the hub for most of the images because since I do have access to it, I won’t get all the prompts contributors get that do not have access to the hub do.

**Comment by argenos** [comment]
> I have added some comments, and while the guide in itself is very good, I am somewhat unsure about the purpose of recommending the github.dev editor to contributors.
> Like this guide tells me how to use the github.dev editor, but it doesn't tell me why I should do it.
> 
Off the top of my head:

Auto-closing of brackets and parenthesis. Editing multiple files, and easier upload of new ones. The file tree is more intuitive to add new files in the right folders, and makes it more noticeable when files don’t have an extension (e.g. the lack of .md). In the future, the ability to use markdownlint. Using preview in a second pane. 
> Like, using the normal GitHub Editor looks more intuitive to less tech-savvy users to me, and the other guide you wrote for that also contains far less references to git concepts (fork, branch, etc.).
> 
> After reading your guide I am still asking myself, what the benefit of using the github.dev editor actually is. There are some things like having code editor commands when editing, but a less tech-savvy contributor might not be missing those anyway, but is confronted with a more complex editor with a lot more buttons and git lingo.
> 

Those concepts are also in the previous guide. We can’t really escape git concepts when contributors are submitting new content through GitHub. It is also my intention to mention them explicit, because that’s needed vocabulary when we discuss changes, reviews or anything else also in Discord.

> And for the more tech-savvy contributors, using the githubdev editor means potentially learning a new tool when they are using a different code editor than VS Code (like me), as opposed to using the normal GitHub editor tech-savvy people are already familiar with. And having things like code editor commands isn't really a plus, since tech-savvy people that do more than mini-contributions would/could also simply use Obsidian (another tool they are already familiar with) and push the vault.
> 
Tech Savy contributors should just do a normal PR in this case. This guide isn’t really meant for them. If they still want to use the editor, I don’t see anything wrong with learning about vscode web.

**Comment by claremacrae** [comment]
> I don’t think the text itself is necessary for all the images, it’s more about where the buttons are. I did zoom into the ones I thought mattered. 

Understood. Personally, in a number of the whole-screen images I can't see the buttons or arrows pointing to them, when viewed at the sort of size they'll be published at... Others may have better eye site or lower res screens, of course.

> I was not using the hub for most of the images because since I do have access to it, I won’t get all the prompts contributors get that do not have access to the hub do.

Yes, I wondered about that. I was thinking of setting a second GitHub account so that I could make screenshots...

**Review by claremacrae — APPROVED** [review]
I'm approve this, because I feel that we should allow hub users to take advantage of the work that argenos has kindly done - and that the docs are definitely good enough to get people started with the new editing UI.

There will be time in future to make further additions and links to it, as we learn and get feedback from users.

And we can add more descriptive text about the pros and cons of the different editing approaches.

File: Editing notes using the github.dev editor.md (hunk 4 above)
Comment (kometenstaub) [inline Editing notes using the github.dev editor.md:15]:
```suggestion
In particular, if you are browsing the Hub and you see a note that you would like to improve, this guide shows you how to make the edit, using the "Edit in GitHub" button at the bottom of the note.
```

File: Editing notes using the github.dev editor.md (hunk 5 above)
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:31]: In VSCode, there is a `+` button to add them all, I wouldn't go into too much detail which could be confusing.

File: Editing notes using the github.dev editor.md (hunk 6 above)
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:36]: It says that also enter can be pressed, so that should be alright.

File: Editing notes using the github.dev editor.md (hunk 7 above)
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:48]:
```suggestion
This is called creating a **Pull Request**.
```

File: Editing notes using the github.dev editor.md (hunk 9 above)
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:54]: It already adds the title from the commit message. If a second or third line was used (I think not possible with this interface), then that would show up there.
And the contributor didn't describe what changes they made before, did they?

File: Editing notes using the github.dev editor.md (hunk 8 above)
Comment (kometenstaub) [inline Editing notes using the github.dev editor.md:61]:
```suggestion
Your edits will later be reviewed by a volunteer of the Hub project, and then merged in to the latest version of the Hub.
Reviewers may request changes before merging. 
```

**Review by kometenstaub — APPROVED** [review]
I didn't review the initial commits, I went through all files in their final version, so some comments I replied to might have been outdated.

File: Editing notes using the github.dev editor.md (hunk 3 above)
Comment (kometenstaub) [reply Editing notes using the github.dev editor.md:51]: If need be, these images can also be opened at full size with right click, open in new window. As you said below, it is something we can improve later.
